src/components/stats/Stats.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+
+import cn from 'classnames'
+import { useDispatch } from 'react-redux'
+
+import { ID } from 'common/models/Identifiers'
+import { formatCount } from 'common/utils/formatUtil'
+import {
+  setUsers,
+  setVisibility
+} from 'store/application/ui/userListModal/slice'
+import {
+  UserListEntityType,
+  UserListType
+} from 'store/application/ui/userListModal/types'
+
+import styles from './Stats.module.css'
+
+type StatProps = {
+  key: 'follower' | 'following'
+  number: number
+  title: string
+  onClick?: () => void
+}
+
+const Stat = ({ number, onClick, title }: StatProps) => {
+  const zero = number === 0
+  const style = {
+    [styles.clickable]: !!onClick && !zero
+  }
+  return (
+    <div
+      className={cn(styles.stat, style)}
+      onClick={!zero ? onClick : () => {}}
+    >
+      <div className={styles.number}>{formatCount(number)}</div>
+      <div className={styles.title}>{title}</div>
+    </div>
+  )
+}
+
+type StatsProps = {
+  userId: ID
+  currentUserId: ID
+  stats: StatProps[]
+  clickable: boolean
+  size: 'medium' | 'large'
+}
+
+const Stats = ({ size, clickable, stats, userId }: StatsProps) => {
+  const dispatch = useDispatch()
+
+  const setModalVisibility = () => dispatch(setVisibility(true))
+  const setFollowerUsers = () =>
+    dispatch(
+      setUsers({
+        userListType: UserListType.FOLLOWER,
+        entityType: UserListEntityType.USER,
+        id: userId
+      })
+    )
+  const setFollowingUsers = () =>
+    dispatch(
+      setUsers({
+        userListType: UserListType.FOLLOWING,
+        entityType: UserListEntityType.USER,
+        id: userId
+      })
+    )
+
+  const sizing = {
+    [styles.medium]: size === 'medium',
+    [styles.large]: size === 'large'
+  }
+
+  const statClick = (stat: StatProps) => {
```

Comment:
maybe `handleClick` or `handleStatClick`

Reply:
thx, i didn't change this code (just moved to ts), but happy to make the change! we should agree on common nomenclature for these things probably. I like `handleX`

Reply:
yeah no worries then, something we can approach holistically later